src/blocks/packages.rs

```diff
@@ -0,0 +1,370 @@
+                        .is_some_and(|regex| apt.has_matching_update(&updates, regex));
+                    warning_vec.push(warning);
+                    critical_vec.push(critical);
+                }
+                PackageManager::Pacman => {
+                    let pacman = Pacman::new();
+                    let updates = pacman.get_updates_list().await?;
+                    pacman_count = pacman.get_update_count(&updates).await?;
+                    let warning = warning_updates_regex
+                        .as_ref()
+                        .is_some_and(|regex| pacman.has_matching_update(&updates, regex));
+                    let critical = critical_updates_regex
+                        .as_ref()
+                        .is_some_and(|regex| pacman.has_matching_update(&updates, regex));
+                    warning_vec.push(warning);
+                    critical_vec.push(critical);
+                }
+                PackageManager::Aur => {
+                    let aur = Aur::new();
+                    let updates = aur.get_updates_list().await?;
+                    aur_count = aur.get_update_count(&updates).await?;
+                    let warning = warning_updates_regex
+                        .as_ref()
+                        .is_some_and(|regex| aur.has_matching_update(&updates, regex));
+                    let critical = critical_updates_regex
+                        .as_ref()
+                        .is_some_and(|regex| aur.has_matching_update(&updates, regex));
+                    warning_vec.push(warning);
+                    critical_vec.push(critical);
+                }
+            }
+        }
+
+        let mut widget = Widget::new();
+
+        let total_count = apt_count + pacman_count + aur_count;
+        widget.set_format(match total_count {
+            0 => format_up_to_date.clone(),
+            1 => format_singular.clone(),
+            _ => format.clone(),
+        });
+        widget.set_values(map!(
+            "icon" => Value::icon("update"),
+            "apt" => Value::number(apt_count),
+            "pacman" => Value::number(pacman_count),
+            "aur" => Value::number(aur_count),
+            "total" => Value::number(total_count),
+        ));
+
+        let warning = warning_vec.iter().any(|&x| x);
+        let critical = critical_vec.iter().any(|&x| x);
+
+        widget.state = match total_count {
+            0 => State::Idle,
+            _ => {
+                if critical {
+                    State::Critical
+                } else if warning {
+                    State::Warning
+                } else {
+                    State::Info
+                }
+            }
+        };
+        api.set_widget(widget)?;
+
+        select! {
+            _ = sleep(config.interval.0) => (),
+            _ = api.wait_for_update_request() => (),
+        }
+    }
+}
+
+#[async_trait]
+trait Backend {
+    async fn setup(&mut self) -> Result<()>;
```

Comment:
There is no reason for `setup()` to be part of this trait. Do the necessary setup work in each backend's constructor.

Reply:
Done